dependencies.yaml

```diff
@@ -312,3 +312,14 @@ dependencies:
               - cuda-nvcc
           - matrix:
             packages:
+    specific:
```

Comment:
```suggestion
```

This second `specific:` is going to clobber the first one.

Try this out:

```python
import yaml
import pprint

with open("dependencies.yaml") as f:
    parsed = yaml.safe_load(f)

pprint.pprint(parsed["dependencies"]["test_python"])
```

You'll see that entry with `cuda-nvcc` is lost.

```text
import yaml
import pprint

with open("dependencies.yaml") as f:
    parsed = yaml.safe_load(f)

pprint.pprint(parsed["dependencies"]["test_python"])
```

This is because:

* `rapids-dependency-file-generator` is written in Python
* the YAML-loading library it uses represents YAML data as a Python dictionary

https://github.com/rapidsai/dependency-file-generator/issues/104 tracks the (not currently active) work to change that. A YAML linter might also be able to detect and prevent this case.

Reply:
Thanks (also for the other tip)!  So one of the jobs shouldn't end up with the constraints, will look into it.
(Over at cudf, I had to notice that it becomes even a bit more annoying to add `cupy` into the mix, since it requires `cupy_cu11` or `cupy_cu12` depending on cuda version.)

Reply:
> since it requires cupy_cu11 or cupy_cu12 depending on cuda version

For sure. I'm very familiar with the requirements for these `dependencies.yaml` files and why the suffixes are split up the way they are (I did https://github.com/rapidsai/cudf/pull/16183 and similar across RAPIDS), so `@` me any time and I'm happy to help with a review.